tests/cli/test_rasa_x.py

```diff
@@ -80,6 +84,101 @@ def test_if_endpoint_config_is_invalid_in_local_mode(kwargs: Dict):
     assert not x._is_correct_event_broker(config)
 
 
+def test_overwrite_model_server_url():
+    """
+    Ensures the model url is overwritten
+    :return:
+    """
+    endpoint_config = EndpointConfig(url="http://testserver:5002/models/default@latest")
+    endpoints = AvailableEndpoints(model=endpoint_config)
+    x._overwrite_endpoints_for_local_x(endpoints, "test", "http://localhost")
+    assert (
+        endpoints.model.url == "http://localhost/projects/default/models/tag/production"
+    )
+    # Check that we get INFO message about overwriting the endpoints configuration
+    log_message = "Ignoring url 'http://testserver:5002/models/default@latest' from 'endpoints.yml' and using 'http://localhost/projects/default/models/tag/production' instead"
+    with warnings.catch_warnings(record=True):
+        x._overwrite_endpoints_for_local_x(endpoints, "test", "http://localhost")
+
+
+def test_reuse_wait_time_between_pulls():
+    """
+    Checks to ensure wait_between_pull_time is honored
+    :return:
+    """
+    test_wait_time = 5
+    endpoint_config = EndpointConfig(
+        url="http://localhost:5002/models/default@latest",
+        wait_time_between_pulls=test_wait_time,
+    )
+    endpoints = AvailableEndpoints(model=endpoint_config)
+    assert endpoints.model.kwargs["wait_time_between_pulls"] == test_wait_time
+
+
+def test_no_wait_time_between_pulls():
+    """
+    Checks to ensure when wait_between_pull_time isn't specified it should be default value
+    :return:
+    """
+    endpoint_config = EndpointConfig(url="http://localhost:5002/models/default@latest")
+    endpoints = AvailableEndpoints(model=endpoint_config)
+    x._overwrite_endpoints_for_local_x(endpoints, "test", "http://localhost")
+    assert endpoints.model.kwargs["wait_time_between_pulls"] == 2
+
+
+def test_no_model_server_url():
```

Comment:
same as above, completely up to you
```suggestion
def test_default_model_server_url():
```